src/liboslexec/llvm_util.cpp

```diff
@@ -1428,37 +1452,41 @@ LLVM_Util::op_bool_to_int (llvm::Value* a)
 
 
 llvm::Value *
-LLVM_Util::op_make_safe_div (TypeDesc type, llvm::Value *a, llvm::Value *b)
+LLVM_Util::op_make_safe_div (llvm::Value *a, llvm::Value *b)
 {
-    if (type.basetype == TypeDesc::FLOAT) {
+    if (a->getType() == type_float() && b->getType() == type_float()) {
         llvm::Value *div = builder().CreateFDiv (a, b);
         llvm::Value *zero = constant (0.0f);
         llvm::Value *iszero = builder().CreateFCmpOEQ (b, zero);
         return builder().CreateSelect (iszero, zero, div);
-    } else {
+    }
+    if (a->getType() == type_int() && b->getType() == type_int()) {
         llvm::Value *div = builder().CreateSDiv (a, b);
         llvm::Value *zero = constant (0);
         llvm::Value *iszero = builder().CreateICmpEQ (b, zero);
         return builder().CreateSelect (iszero, zero, div);
     }
+    ASSERT (0 && "Op has bad value type combination");
 }
 
 
 
 llvm::Value *
-LLVM_Util::op_make_safe_mod (TypeDesc type, llvm::Value *a, llvm::Value *b)
+LLVM_Util::op_make_safe_mod (llvm::Value *a, llvm::Value *b)
 {
-    if (type.basetype == TypeDesc::FLOAT) {
+    if (a->getType() == type_float() && b->getType() == type_float()) {
         llvm::Value *mod = builder().CreateFRem (a, b);
         llvm::Value *zero = constant (0.0f);
         llvm::Value *iszero = builder().CreateFCmpOEQ (b, zero);
         return builder().CreateSelect (iszero, zero, mod);
```

Comment:
Right now, this function is not called by OSL, and it is "safe result" but not "SIGFPE-safe". Maybe I should remove it entirely.